## TRAVAIL SUR EAU DU ROBINET 

In [1]:
import pandas as pd

pd.set_option("display.max_columns", None)  # show all cols
pd.set_option("display.max_colwidth", None)  # show full width of showing cols
pd.set_option("display.expand_frame_repr", False)

In [2]:
edr_cap_plv = pd.read_csv(
    r"C:\Users\EPPINGER Muriel\Documents\D4G_13\EDR\CAP_PLV_202412.txt", delimiter=","
)
edr_cap_plv.head(2)

distrcode                                                   distrlib    distrsiret cddept inseecommune nomcommune   cdreseau          nomreseau codetypeinstallation nomtypeinstallation inae    cdpointsurv             nompointsurv referenceprel    dateprel heureprel finaliteprel                                                                                            conclusionprel cdtypeeau              libtypeeau plvconformitebacterio plvconformitechimique plvcomplet plvcdorganismepayeur typevisite plvcondition       preleveur   distrsiren plvconformiterefbacterio plvconformiterefchimique  cddeptsandre dtautorisation champcaptant    codebrgm codebss     dmajplv   dmodifplv   dmodifins representativite  coord_x  coord_y
0  00101028  SERA - SYNDICAT DES EAUX DE LA REGION D'AMBERIEU-EN-BUGEY  2.501018e+13    001        01007   AMBRONAY  001000060  PUITS DU BELLATON                  CAP             CAPTAGE  ESO  0010000000062  PUITS DU BELLATON (SUD)   00100143315  2024-11-25     09h52           CS  Eau brute souterraine conforme aux limites de qualité en vigueur pour l'ensemble des paramètres mesurés.        B   EAU BRUTE SOUTERRAINE                      C                     C          O             00100452  RP                 NaN  13000807100107  250101839.0                        C                        C             1     1967-09-06            N  06754X0040     NaN  2024-12-10  2024-12-11  2025-01-06                O      NaN      NaN
1  00101002                                                SOGEDO LYON  3.011928e+13    001        01027      BALAN  001000015     PUITS DE BALAN                  CAP             CAPTAGE  ESO  0010000000017           PUITS DE BALAN   00100143340  2024-11-20     09h50           CS  Eau brute souterraine conforme aux limites de qualité en vigueur pour l'ensemble des paramètres mesurés.        B   EAU BRUTE SOUTERRAINE                      S                     C          O             00101002  RP                 NaN  13000807100107  301192803.0                        S                        C             1     1988-04-22            N  06991X0179      S2  2024-12-03  2024-12-04  2024-01-15                O      NaN      NaN

In [3]:
edr_cap_plv.describe(include="all")

distrcode distrlib    distrsiret cddept inseecommune nomcommune   cdreseau                            nomreseau codetypeinstallation nomtypeinstallation  inae    cdpointsurv nompointsurv referenceprel    dateprel heureprel finaliteprel                                                                                            conclusionprel cdtypeeau              libtypeeau plvconformitebacterio plvconformitechimique plvcomplet plvcdorganismepayeur typevisite plvcondition       preleveur    distrsiren plvconformiterefbacterio plvconformiterefchimique  cddeptsandre dtautorisation champcaptant    codebrgm codebss     dmajplv   dmodifplv   dmodifins representativite  coord_x  coord_y
count       2720     2720  1.924000e+03   2720         2720       2720       2720                                 2720                 2720                2720  2720           2720         2720          2720        2720      2720         2720                                                                                                      2720      2720                    2720                  2719                  2719       2720                 2709       2720          346            1374  1.924000e+03                     2719                     2719   2720.000000           1893         2057        2216    1791        2720        2720        2720             2720      0.0      0.0
unique       764      701           NaN     97         1690       1683       2202                                 2194                    2                   2     3           2229         2047          2720         137       422            9                                                                                                       331         7                       7                     3                     4          1                  781         35            9              47           NaN                        3                        3           NaN           1098            2        1873     292          42          41         815                1      NaN      NaN
top     08600612     SAUR           NaN    029        10315     RANCES  010000278  RANCES MAIZIERES LES BRIENNE FORAGE                  CAP             CAPTAGE   ESO  0100000000537      EXHAURE   97400140263  2024-12-02     10h30           CS  Eau brute souterraine conforme aux limites de qualité en vigueur pour l'ensemble des paramètres mesurés.        B   EAU BRUTE SOUTERRAINE                      S                     C          O             08600612  RP                 INF  41054531300018           NaN                        S                        C           NaN     2017-01-31            N  02637X0035  HY      2024-12-17  2024-12-16  2023-03-02                O      NaN      NaN
freq         101      108           NaN    157           24         24         12                                   12                 2535                2535  1985             12           69             1         122        41         2443                                                                                                      1400      1956                    1956                  1414                  2611       2720                  101       1374          175             210           NaN                     1414                     2694           NaN             18         1832          12     266         148         158          72             2720      NaN      NaN
mean         NaN      NaN  4.272712e+13    NaN          NaN        NaN        NaN                                  NaN                  NaN                 NaN   NaN            NaN          NaN           NaN         NaN       NaN          NaN                                                                                                       NaN       NaN                     NaN                   NaN                   NaN        NaN                  NaN        NaN          NaN             NaN  4.272712e+08               

In [4]:
edr_cap_res = pd.read_csv(
    r"C:\Users\EPPINGER Muriel\Documents\D4G_13\EDR\CAP_RES_202412.txt", delimiter=","
)
edr_cap_res.head(2)

C:\Users\EPPINGER Muriel\AppData\Local\Temp\ipykernel_3200\4029363174.py:1: DtypeWarning: Columns (0,1,2,5,8) have mixed types. Specify dtype option on import or set low_memory=False.
  edr_cap_res = pd.read_csv(r'C:\Users\EPPINGER Muriel\Documents\D4G_13\EDR\CAP_RES_202412.txt', delimiter=',')


cddept referenceprel    cdanalyse refechantillonlabo datereceptionechant cdlaboratoire   laboratoire cdtypeanalyse analysecdorganisme analysedate insituana  cdparametre cdparametresiseeaux   rqana rssigne  rsana cdunitereferencesiseeaux cdunitereference  cdfractionanalyse  sandrelaboratoire representativite
0      1     100141009  100149900.0      LSE2406-32536          2024-12-26        6901.0  4.105453e+13        0011RP           100973.0  2025-01-14         L       1141.0                 24D  <0,020       <    0.0                     µg/L              133                NaN              217.0                O
1      1     100141009  100149900.0      LSE2406-32536          2024-12-26        6901.0  4.105453e+13        0011RP           100973.0  2025-01-14         L       2011.0               26DCB  <0,005       <    0.0                     µg/L              133                NaN              217.0                O

In [5]:
edr_cap_res.describe(include="all")

cddept  referenceprel     cdanalyse refechantillonlabo datereceptionechant  cdlaboratoire   laboratoire cdtypeanalyse  analysecdorganisme analysedate insituana    cdparametre cdparametresiseeaux   rqana rssigne          rsana cdunitereferencesiseeaux cdunitereference  cdfractionanalyse  sandrelaboratoire representativite
count   426615.0   4.266150e+05  4.167670e+05             416767              414403       416767.0  4.167670e+05        411250            416767.0      415711    426615  425942.000000              425648  426600  376655  425904.000000                   426613           426613               12.0       3.710900e+05           426615
unique      99.0   2.721000e+03  2.911000e+03               2907                 138           49.0           NaN           455               791.0         150         2            NaN                1350    5490       8            NaN                       41               41                NaN                NaN                1
top         75.0   7.700249e+09  7.700249e+09      LSE2411-25855          2024-11-20         6901.0           NaN        075RP7           7500034.0  2024-12-10         L            NaN                  PH  <0,005       <            NaN                     µg/L              133                NaN                NaN                O
freq     24386.0   6.750000e+02  6.700000e+02                670               18567       154208.0           NaN         19173             24222.0       22510    416767            NaN                2552  142888  375766            NaN                   377030           377030                NaN                NaN           426615
mean         NaN            NaN           NaN                NaN                 NaN            NaN  3.100525e+13           NaN                 NaN         NaN       NaN    2595.732144                 NaN     NaN     NaN       7.887604                      NaN              NaN               23.0       2.765410e+12              NaN
std          NaN            NaN           NaN                NaN                 NaN            NaN  1.524719e+13           NaN                 NaN         NaN       NaN    2043.104516                 NaN     NaN     NaN     239.931429                      NaN              NaN                0.0       5.700175e+12              NaN
min          NaN            NaN           NaN                NaN                 NaN            NaN  1.300021e+13           NaN                 NaN         NaN       NaN       0.000000                 NaN     NaN     NaN     -17.200000                      NaN              NaN               23.0       9.800000e+01              NaN
25%          NaN            NaN           NaN                NaN                 NaN            NaN  1.300190e+13           NaN                 NaN         NaN       NaN    1301.000000                 NaN     NaN     NaN       0.000000                      NaN              NaN               23.0       2.110000e+02              NaN
50%          NaN            NaN           NaN                NaN                 NaN            NaN  4.105453e+13           NaN                 NaN         NaN       NaN    1704.000000                 NaN     NaN     NaN       0.000000                      NaN              NaN               23.0       2.170000e+02              NaN
75%          NaN            NaN           NaN                NaN                 NaN            NaN  4.105453e+13           NaN                 NaN         NaN       NaN    2737.750000                 NaN     NaN     NaN       0.000000                      NaN              NaN               23.0       5.860000e+02              NaN
max          NaN            NaN           NaN                NaN                 NaN            NaN  7.788520e+13           NaN                 NaN         NaN       NaN   99999.000000                 NaN     NaN     NaN   52000.000000                      NaN              NaN               23.0       2.000132e+13              NaN

In [6]:
# modification format reference prel pour comparaison avec autres bdd

edr_cap_res["referenceprel"] = edr_cap_res["referenceprel"].astype("string")
edr_cap_res["referenceprel"] = edr_cap_res["referenceprel"].apply(
    lambda x: "00" + x if len(x) == 9 else "0" + x if len(x) == 10 else x
)

Vérification faite le prélèvement '00100143315' n'est pas présent dans la base EDR_UDI

In [7]:
# recherche de doublons dans la base des prélevements

doublons = edr_cap_plv.duplicated()
df_doublons_plv = edr_cap_plv[doublons]
print(df_doublons_plv)

# recherche de doublons dans la base des résultats

doublons = edr_cap_res.duplicated()
df_doublons_res = edr_cap_res[doublons]
print(df_doublons_res)

Empty DataFrame
Columns: [distrcode, distrlib, distrsiret, cddept, inseecommune, nomcommune, cdreseau, nomreseau, codetypeinstallation, nomtypeinstallation, inae, cdpointsurv, nompointsurv, referenceprel, dateprel, heureprel, finaliteprel, conclusionprel, cdtypeeau, libtypeeau, plvconformitebacterio, plvconformitechimique, plvcomplet, plvcdorganismepayeur, typevisite, plvcondition, preleveur, distrsiren, plvconformiterefbacterio, plvconformiterefchimique, cddeptsandre, dtautorisation, champcaptant, codebrgm, codebss, dmajplv, dmodifplv, dmodifins, representativite, coord_x, coord_y]
Index: []
Empty DataFrame
Columns: [cddept, referenceprel, cdanalyse, refechantillonlabo, datereceptionechant, cdlaboratoire, laboratoire, cdtypeanalyse, analysecdorganisme, analysedate, insituana, cdparametre, cdparametresiseeaux, rqana, rssigne, rsana, cdunitereferencesiseeaux, cdunitereference, cdfractionanalyse, sandrelaboratoire, representativite]
Index: []


In [8]:
# création d'une liste unique de referenceprel

edr_plv = edr_cap_plv[["referenceprel"]].copy()
unique_edr_cap_plv = edr_plv.drop_duplicates()
unique_edr_cap_plv.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2720 entries, 0 to 2719
Data columns (total 1 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   referenceprel  2720 non-null   object
dtypes: object(1)
memory usage: 21.4+ KB


## TRAVAIL SUR EAU DE COMMUNE

In [9]:
import duckdb
from pipelines.tasks._common import DUCKDB_FILE

con = duckdb.connect(database=DUCKDB_FILE, read_only=True)

In [10]:
query_2024 = """
SELECT * from edc_prelevements
"""

edc_plv = con.sql(query_2024)
data_edc_plv = edc_plv.df()
data_edc_plv.head(2)

cddept   cdreseau inseecommuneprinc nomcommuneprinc cdreseauamont      nomreseauamont pourcentdebit referenceprel   dateprel heureprel                                                                                           conclusionprel                         ugelib                        distrlib                          moalib plvconformitebacterio plvconformitechimique plvconformitereferencebact plvconformitereferencechim  de_partition de_ingestion_date de_dataset_datetime
0    001  001000003             01007        AMBRONAY          None                None          None   00100119766 2020-02-13     11h40  Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.  SI REGION D'AMBERIEU-EN-BUGEY  SIE REGION D'AMBERIEU-EN-BUGEY  SIE REGION D'AMBERIEU-EN-BUGEY                     C                     C                          C                          C          2020        2025-02-14     20230811-150005
1    001  001000003             01007        AMBRONAY     001001304  TTP (CLG) AMBRONAY         100 %   00100120290 2020-03-17     11h15  Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.  SI REGION D'AMBERIEU-EN-BUGEY  SIE REGION D'AMBERIEU-EN-BUGEY  SIE REGION D'AMBERIEU-EN-BUGEY                     C                     C                          C                          C          2020        2025-02-14     20230811-150005

In [11]:
edc_plv = data_edc_plv[["referenceprel"]].copy()
unique_edc_plv = edc_plv.drop_duplicates()
unique_edc_plv.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1436114 entries, 0 to 2083344
Data columns (total 1 columns):
 #   Column         Non-Null Count    Dtype 
---  ------         --------------    ----- 
 0   referenceprel  1436114 non-null  object
dtypes: object(1)
memory usage: 21.9+ MB


In [21]:
# extraction d'une referenceprel dnas bdd "prélèvements"

edr_cap_plv.loc[edr_cap_plv["referenceprel"] == "06400191643", :]

distrcode        distrlib    distrsiret cddept inseecommune nomcommune   cdreseau  nomreseau codetypeinstallation nomtypeinstallation inae    cdpointsurv      nompointsurv referenceprel    dateprel heureprel finaliteprel                                                                                                                                                                                                                                                                                                                        conclusionprel cdtypeeau                        libtypeeau plvconformitebacterio plvconformitechimique plvcomplet plvcdorganismepayeur typevisite plvcondition preleveur   distrsiren plvconformiterefbacterio plvconformiterefchimique  cddeptsandre dtautorisation champcaptant    codebrgm codebss     dmajplv   dmodifplv   dmodifins representativite  coord_x  coord_y
1996  06401336  CA PAYS BASQUE  2.000671e+13    064        64259    HELETTE  064000255  OYAMBURUA                  CAP             CAPTAGE  ESO  0640000001377  SOURCE OYAMBURUA   06400191643  2020-07-29     11h30           CS  Eau d'alimentation conforme aux limites de qualité et non conforme aux références de qualité. Conductivité inférieure à la valeur minimale de référence. Eau peu minéralisée, favorisant la dégradation des installations (canalisations ...) et pouvant induire la dissolution d'éléments indésirables (plomb, cuivre, nickel ...).        S   EAU DISTRIBUEE SANS DESINFECTION                     C                     C          O             06401336  P2                 PNF       NaN  200067106.0                        C                        N            64     1994-10-07          NaN  10272X0005  ERH     2024-12-17  2024-12-18  2023-11-30                O      NaN      NaN

In [22]:
exple = """
SELECT * from edc_prelevements
WHERE referenceprel='06400191643'
"""

edc_plv_exple = con.sql(exple)
data_edc_plv_exple = edc_plv_exple.df()
data_edc_plv_exple

cddept   cdreseau inseecommuneprinc nomcommuneprinc cdreseauamont nomreseauamont pourcentdebit referenceprel   dateprel heureprel                                                                                                                                                                                                                                                                                                                        conclusionprel              ugelib        distrlib          moalib plvconformitebacterio plvconformitechimique plvconformitereferencebact plvconformitereferencechim  de_partition de_ingestion_date de_dataset_datetime
0    064  064000506             64259         HELETTE     064000255      OYAMBURUA          None   06400191643 2020-07-29     11h30  Eau d'alimentation conforme aux limites de qualité et non conforme aux références de qualité. Conductivité inférieure à la valeur minimale de référence. Eau peu minéralisée, favorisant la dégradation des installations (canalisations ...) et pouvant induire la dissolution d'éléments indésirables (plomb, cuivre, nickel ...).  CA PAYS BASQUE (1)  CA PAYS BASQUE  CA PAYS BASQUE                     C                     C                          C                          N          2020        2025-02-14     20230811-150005

In [23]:
# extraction d'un prélèvement dans bdd resultats

edr_cap_res.loc[edr_cap_res["referenceprel"] == "06400191643", :]

cddept referenceprel     cdanalyse refechantillonlabo datereceptionechant cdlaboratoire   laboratoire cdtypeanalyse analysecdorganisme analysedate insituana  cdparametre cdparametresiseeaux   rqana rssigne  rsana cdunitereferencesiseeaux cdunitereference  cdfractionanalyse  sandrelaboratoire representativite
286141     64   06400191643  6400191595.0             769939          2020-07-29        6401.0  4.188141e+13       064NP2A          6401336.0  2020-08-18         L       1284.0              111TCL    <0,5       <   0.00                     µg/L              133                NaN              586.0                O
286142     64   06400191643  6400191595.0             769939          2020-07-29        6401.0  4.188141e+13       064NP2A          6401336.0  2020-08-18         L       1271.0             1122TCE      <1       <   0.00                     µg/L              133                NaN              586.0                O
286143     64   06400191643  6400191595.0             769939          2020-07-29        6401.0  4.188141e+13       064NP2A          6401336.0  2020-08-18         L       1285.0              112TCE    <0,5       <   0.00                     µg/L              133                NaN              586.0                O
286144     64   06400191643  6400191595.0             769939          2020-07-29        6401.0  4.188141e+13       064NP2A          6401336.0  2020-08-18         L       1162.0              11DCET      <1       <   0.00                     µg/L              133                NaN              586.0                O
286145     64   06400191643  6400191595.0             769939          2020-07-29        6401.0  4.188141e+13       064NP2A          6401336.0  2020-08-18         L       1160.0              11DCLE      <1       <   0.00                     µg/L              133                NaN              586.0                O
...       ...           ...           ...                ...                 ...           ...           ...           ...                ...         ...       ...          ...                 ...     ...     ...    ...                      ...              ...                ...                ...              ...
286366     64   06400191643  6400191595.0             769939          2020-07-29        6401.0  4.188141e+13       064NP2A          6401336.0  2020-08-18         L       1719.0               TLFND  <0,005       <   0.00                     µg/L              133                NaN              586.0                O
286367     64   06400191643  6400191595.0             769939          2020-07-29        6401.0  4.188141e+13       064NP2A          6401336.0  2020-08-18         L       1289.0                TRIF  <0,005       <   0.00                     µg/L              133                NaN              586.0                O
286368     64   06400191643  6400191595.0             769939          2020-07-29        6401.0  4.188141e+13       064NP2A          6401336.0  2020-08-18         L       1295.0             TURBNFU    0,29     NaN   0.29                      NFU              232                NaN              586.0                O
286369     64   06400191643  6400191595.0             769939          2020-07-29        6401.0  4.188141e+13       064NP2A          6401336.0  2020-08-18         L       1291.0                VCLZ  <0,005       <   0.00                     µg/L              133                NaN              586.0                O
286370     64   06400191643  6400191595.0             769939          2020-07-29        6401.0  4.188141e+13       064NP2A          6401336.0  2020-08-18         L       1290.0                VMDT   <0,02       <   0.00                     µg/L              133                NaN              586.0                O

[230 rows x 21 columns]

In [24]:
exple_res = """
SELECT * from edc_resultats
WHERE referenceprel='06400191643'
"""

edc_res_exple = con.sql(exple_res)
data_edc_res_exple = edc_res_exple.df()
data_edc_res_exple

cddept referenceprel cdparametresiseeaux  cdparametre                libmajparametre                libminparametre libwebparametre qualitparam insituana   rqana cdunitereferencesiseeaux cdunitereference  limitequal  refqual  valtraduite    casparam referenceanl  de_partition de_ingestion_date de_dataset_datetime
0      064   06400191643              111TCL         1284          TRICHLOROÉTHANE-1,1,1          Trichloroéthane-1,1,1            None           N         L    <0,5                     µg/L              133        None     None         0.00     71-55-6  06400191595          2020        2025-02-14     20230811-150005
1      064   06400191643             1122TCE         1271      TÉTRACHLOROÉTHANE-1,1,2,2      Tétrachloroéthane-1,1,2,2            None           N         L      <1                     µg/L              133        None     None         0.00     79-34-5  06400191595          2020        2025-02-14     20230811-150005
2      064   06400191643              112TCE         1285          TRICHLOROÉTHANE-1,1,2          Trichloroéthane-1,1,2            None           N         L    <0,5                     µg/L              133        None     None         0.00     79-00-5  06400191595          2020        2025-02-14     20230811-150005
3      064   06400191643              11DCET         1162           DICHLOROÉTHYLÈNE-1,1           Dichloroéthylène-1,1            None           N         L      <1                     µg/L              133        None     None         0.00     75-35-4  06400191595          2020        2025-02-14     20230811-150005
4      064   06400191643              11DCLE         1160             DICHLOROÉTHANE-1,1             Dichloroéthane-1,1            None           N         L      <1                     µg/L              133        None     None         0.00     75-34-3  06400191595          2020        2025-02-14     20230811-150005
..     ...           ...                 ...          ...                            ...                            ...             ...         ...       ...     ...                      ...              ...         ...      ...          ...         ...          ...           ...               ...                 ...
225    064   06400191643               TLFND         1719                  TOLYLFLUANIDE                  Tolylfluanide            None           N         L  <0,005                     µg/L              133  <=0,1 µg/L     None         0.00    731-27-1  06400191595          2020        2025-02-14     20230811-150005
226    064   06400191643                TRIF         1289                   TRIFLURALINE                   Trifluraline            None           N         L  <0,005                     µg/L              133  <=0,1 µg/L     None         0.00   1582-09-8  06400191595          2020        2025-02-14     20230811-150005
227    064   06400191643             TURBNFU         1295  TURBIDITÉ NÉPHÉLOMÉTRIQUE NFU  Turbidité néphélométrique NFU            None           N         L    0,29                      NFU              232        None  <=2 NFU         0.29        None  06400191595          2020        2025-02-14     20230811-150005
228    064   06400191643                VCLZ         1291                  VINCHLOZOLINE                  Vinchlozoline            None           N         L  <0,005                     µg/L              133  <=0,1 µg/L     None         0.00  50471-44-8  06400191595          2020        2025-02-14     20230811-150005
229    064   06400191643                VMDT         1290                    VAMIDOTHION                    Vamidothion            None           N         L   <0,02                     µg/L              133  <=0,1 µg/L     None         0.00   2275-23-2  06400191595          2020        2025-02-14     20230811-150005

[230 rows x 20 columns]

In [25]:
# recherche des prélèvements présents dans EDR et absents dans EDC

resultat = unique_edr_cap_plv[
    ~unique_edr_cap_plv["referenceprel"].isin(unique_edc_plv["referenceprel"])
]
print(resultat)

     referenceprel
0      00100143315
1      00100143340
2      00100143335
3      00100141972
4      00100143304
...            ...
2715   97400140607
2716   97400140814
2717   97400140608
2718   97400140597
2719   97400140495

[2716 rows x 1 columns]


In [26]:
edr_cap_plv.loc[edr_cap_plv["referenceprel"] == "00100143340", :]

distrcode     distrlib    distrsiret cddept inseecommune nomcommune   cdreseau       nomreseau codetypeinstallation nomtypeinstallation inae    cdpointsurv    nompointsurv referenceprel    dateprel heureprel finaliteprel                                                                                            conclusionprel cdtypeeau              libtypeeau plvconformitebacterio plvconformitechimique plvcomplet plvcdorganismepayeur typevisite plvcondition       preleveur   distrsiren plvconformiterefbacterio plvconformiterefchimique  cddeptsandre dtautorisation champcaptant    codebrgm codebss     dmajplv   dmodifplv   dmodifins representativite  coord_x  coord_y
1  00101002  SOGEDO LYON  3.011928e+13    001        01027      BALAN  001000015  PUITS DE BALAN                  CAP             CAPTAGE  ESO  0010000000017  PUITS DE BALAN   00100143340  2024-11-20     09h50           CS  Eau brute souterraine conforme aux limites de qualité en vigueur pour l'ensemble des paramètres mesurés.        B   EAU BRUTE SOUTERRAINE                      S                     C          O             00101002  RP                 NaN  13000807100107  301192803.0                        S                        C             1     1988-04-22            N  06991X0179      S2  2024-12-03  2024-12-04  2024-01-15                O      NaN      NaN

In [29]:
exple = """
SELECT * from edc_prelevements
WHERE inseecommuneprinc='01027' and dateprel='2024-11-20'
"""

edc_plv_exple = con.sql(exple)
data_edc_plv_exple = edc_plv_exple.df()
data_edc_plv_exple

cddept   cdreseau inseecommuneprinc nomcommuneprinc cdreseauamont   nomreseauamont pourcentdebit referenceprel   dateprel heureprel                                                                                           conclusionprel                              ugelib     distrlib                             moalib plvconformitebacterio plvconformitechimique plvconformitereferencebact plvconformitereferencechim  de_partition de_ingestion_date de_dataset_datetime
0    001  001000623             01027           BALAN          None             None          None   00100143187 2024-11-20     13h48  Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.  CC COTIERE A MONTLUEL (3CM) SOGEDO  SOGEDO LYON  CC DE LA CÔTIÈRE À MONTLUEL (3CM)                     C                     C                          C                          C          2024        2025-02-14     20250203-075802
1    001  001000623             01027           BALAN     001000619  TTP (CLG) BALAN         100 %   00100143240 2024-11-20     11h44  Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.  CC COTIERE A MONTLUEL (3CM) SOGEDO  SOGEDO LYON  CC DE LA CÔTIÈRE À MONTLUEL (3CM)                     C                     C                          C                          C          2024        2025-02-14     20250203-075802
2    001  001000623             01027           BALAN     001000619  TTP (CLG) BALAN         100 %   00100143338 2024-11-20     11h44  Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.  CC COTIERE A MONTLUEL (3CM) SOGEDO  SOGEDO LYON  CC DE LA CÔTIÈRE À MONTLUEL (3CM)                     S                     C                          S                          C          2024        2025-02-14     20250203-075802

CONCLUSION : 

La base EDR CAP 2024 contient 2720 prélèvements 

2716 de ces prélèvements sont absents de la base EDC (soit 99.8%). 

Des vérifications on été faites sur inseecommune/date/heure pour qq prélèvements pour pallier à une anomalie sur la référence. 

Aucune correspondance n'a été détectée. 

Réfléchir sur l'intérêt d'étendre ces résultats sur la totalité des années 

Réfléchir sur l'intérêt d'intégrer ces informations dans la base EDC